api/src/main/java/edu/cornell/mannlib/vitro/webapp/dynapi/DynamicAPIDocumentation.java

```diff
@@ -1,5 +1,770 @@
+        requestBody.setContent(content);
+
+        operation.setRequestBody(requestBody);
+
+        ApiResponses apiResponses = new ApiResponses();
+
+        addOkApiResponse(apiResponses, action);
+        addUnauthorizedApiResponse(apiResponses);
+        addForbiddenApiResponse(apiResponses);
+        addNotFoundApiResponse(apiResponses);
+
+        operation.setResponses(apiResponses);
+
+        return operation;
+    }
+
+    private Operation individualDeleteOperation(Action action, Tag tag) {
+        Operation operation = new Operation();
+        operation.addTagsItem(tag.getName());
+
+        // No description available per resource API rpc
+        // operation.setDescription("");
+
+        ApiResponses apiResponses = new ApiResponses();
+
+        addOkApiResponse(apiResponses, action);
+        addUnauthorizedApiResponse(apiResponses);
+        addForbiddenApiResponse(apiResponses);
+        addNotFoundApiResponse(apiResponses);
+
+        operation.setResponses(apiResponses);
+
+        return operation;
+    }
+
+    private Operation customRESTActionGetOperation(Action action, Tag tag) {
+        return collectionGetOperation(action, tag);
+    }
+
+    private Operation customRESTActionPostOperation(Action action, Tag tag) {
+        return collectionPostOperation(action, tag);
+    }
+
+    private Operation customRESTActionPutOperation(Action action, Tag tag) {
+        return individualPutOperation(action, tag);
+    }
+
+    private Operation customRESTActionPatchOperation(Action action, Tag tag) {
+        return individualPatchOperation(action, tag);
+    }
+
+    private Operation customRESTActionDeleteOperation(Action action, Tag tag) {
+        return individualDeleteOperation(action, tag);
+    }
+
+    private Operation actionPostOperation(Action action, Tag tag) {
+        Operation operation = new Operation();
+        operation.addTagsItem(tag.getName());
+
+        // No description available per resource API rpc
+        // operation.setDescription("");
+
+        RequestBody requestBody = new RequestBody();
+        Content content = new Content();
+        MediaType mediaType = new MediaType();
+        ObjectSchema schema = new ObjectSchema();
+
+        buildObjectSchema(schema, action.getRequiredParams());
+
+        mediaType.schema(schema);
+        content.addMediaType("application/json", mediaType);
+        requestBody.setContent(content);
+
+        operation.setRequestBody(requestBody);
+
+        ApiResponses apiResponses = new ApiResponses();
+
+        addOkApiResponse(apiResponses, action);
+        addUnauthorizedApiResponse(apiResponses);
+        addForbiddenApiResponse(apiResponses);
+        addNotFoundApiResponse(apiResponses);
+
+        operation.setResponses(apiResponses);
+
+        return operation;
+    }
+
+    private PathParameter individualPathParameter() {
+        PathParameter pathParameter = new PathParameter();
+
+        pathParameter.setName("resourceId");
```

Comment:
maybe to define a constant, or to use this one https://github.com/vivo-project/Vitro/blob/sprint-dynapi-2022-feb-staging/api/src/main/java/edu/cornell/mannlib/vitro/webapp/dynapi/OperationData.java#L18 (you can change access to public and value to "resourceId").